backend/uclapi/roombookings/management/commands/update_gencache.py

```diff
@@ -35,53 +36,70 @@ def handle(self, *args, **options):
 
         self.stdout.write("Selecting a clone table...")
         lock = Lock.objects.all()[0]
-        curr = BookingA if lock.bookingA else BookingB
+        current_bucket = BookingA if lock.bookingA else BookingB
 
         self.stdout.write("Flushing the clone table...")
         # flush the table
         cursor = connections['gencache'].cursor()
         cursor.execute(
             "TRUNCATE TABLE {} RESTART IDENTITY;".format(
-                    "roombookings_" + curr.__name__.lower()))
+                    "roombookings_" + current_bucket.__name__.lower()))
 
         self.stdout.write(
             "Dumping all the data from Oracle into a new list..."
         )
-        data_objects = []
-
-        for row in cur:
-            data_objects.append(curr(
-                setid=row[0],
-                siteid=row[1],
-                roomid=row[2],
-                sitename=row[3],
-                roomname=row[4],
-                bookabletype=row[5],
-                slotid=row[6],
-                bookingid=row[7],
-                starttime=row[8],
-                finishtime=row[9],
-                startdatetime=row[10],
-                finishdatetime=row[11],
-                weeknumber=row[12],
-                condisplayname=row[13],
-                phone=row[14],
-                descrip=row[15],
-                title=row[16]
-            ))
-
-        self.stdout.write("There are " + str(len(data_objects)) + " records.")
-
-        self.stdout.write("Bulk creating this in PostgreSQL...")
-        curr.objects.using("gencache").bulk_create(
-            data_objects,
-            batch_size=5000
-        )
+
+        batch_size = 5000
+        running_total = 0
+
+        while True:
+            data_objects = []
+            for row in cur.fetchmany(numRows=batch_size):
+                data_objects.append(current_bucket(
```

Comment:
Can't this be `current_bucket(*row)`? @Faiz

Reply:
I'm not sure it can be because we match the fields to Django model definitions. Plus, why `*row` and not `**row`? Perhaps that my Python knowledge lacking...

Reply:
> we match the fields to Django model definitions

True, have not considered that. (I thought the model definition has it in the order we specify it)

> why `*row` and not `**row`? 

`**` should be used with mappings (e.g. dict); `fetchmany(...)` returns a tuple which you access using indices therefore `*`.

Reply:
My only concern is that if the underlying table or view structure changes (even with the addition of just one more field) the code will error out. At the moment we are fine unless data is removed, but if data is added nothing will break. If that changes in the future (e.g. if UCL do another join in our view's code) then doing `**row` will break the caching.

Reply:
(I love the Pythonic magic of the double asterisk though! I'm very conflicted on this)